tests/components/pushover/test_notify.py

```diff
@@ -0,0 +1,77 @@
+"""Test the pushover notify platform."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components.pushover import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=False)
+def mock_pushover():
+    """Mock pushover."""
+    with patch(
+        "pushover_complete.PushoverAPI._generic_post", return_value={}
+    ) as mock_generic_post:
+        yield mock_generic_post
+
+
+@pytest.fixture
+def mock_send_message():
+    """Patch PushoverAPI.send_message for TTL test."""
+    with patch(
+        "homeassistant.components.pushover.notify.PushoverAPI.send_message"
+    ) as mock:
+        yield mock
+
+
+async def test_send_message_with(
+    hass: HomeAssistant, mock_pushover: MagicMock, mock_send_message: MagicMock
+) -> None:
+    """Test sending a message with TTL included in the data."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "name": "pushover",
+            "api_key": "API_KEY",
+            "user_key": "USER_KEY",
+        },
+    )
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        "notify",
+        "pushover",
+        {"message": "Hello TTL", "data": {"ttl": 900}},
+        blocking=True,
+    )
+
+    assert mock_send_message.called
+
+    call_args = mock_send_message.call_args.kwargs
+    assert call_args["message"] == "Hello TTL"
+    assert call_args["ttl"] == 900
+
```

Comment:
This is checked below by `assert_called_once_with`, we don't need to check it here too
```suggestion
```